internal/pkg/agent/configuration/settings.go

```diff
@@ -16,7 +17,7 @@ import (
 )
 
 // ExternalInputsPattern is a glob that matches the paths of external configuration files.
-var ExternalInputsPattern = filepath.Join("inputs.d", "*.yml")
```

Comment:
You should not use `paths.AgentInputsDPath()` at initialization. It is possible that it will change here: https://github.com/elastic/elastic-agent/blob/main/internal/pkg/agent/cmd/container.go#L852 because it calls `setPaths`.


So you should have the code path that does the searching perform the join at that point, not here in a global var.

Reply:
👍 good catch

Reply:
good catch indeed.